tests/v2/validation/rbac/rbac_test.go

```diff
@@ -454,47 +496,76 @@ func (rb *RBTestSuite) TestRBAC() {
 }
 
 func (rb *RBTestSuite) TestRBACAdditional() {
-	rb.Run("Set up User with cluster Role for additional rbac test cases "+roleOwner, func() {
-		newUser, err := createUser(rb.client, standardUser)
-		require.NoError(rb.T(), err)
-		rb.standardUser = newUser
-		rb.T().Logf("Created user: %v", rb.standardUser.Username)
-		rb.standardUserClient, err = rb.client.AsUser(newUser)
-		require.NoError(rb.T(), err)
 
-		subSession := rb.session.NewSession()
-		defer subSession.Cleanup()
+	tests := []struct {
+		name   string
+		member string
+	}{
+		{"Standard User", standardUser},
+		{"Restricted Admin", restrictedAdmin},
+	}
 
-		rb.T().Logf("Adding user as " + roleOwner + " to the downstream cluster.")
-		//Adding created user to the downstream clusters with the role cluster Owner.
-		err = users.AddClusterRoleToUser(rb.client, rb.cluster, rb.standardUser, roleOwner)
-		require.NoError(rb.T(), err)
-		rb.standardUserClient, err = rb.standardUserClient.ReLogin()
-		require.NoError(rb.T(), err)
+	for _, tt := range tests {
+		rb.Run("Set up User with cluster Role for additional rbac test cases "+roleOwner, func() {
+			newUser, err := createUser(rb.client, tt.member)
+			require.NoError(rb.T(), err)
+			rb.standardUser = newUser
+			rb.T().Logf("Created user: %v", rb.standardUser.Username)
+			rb.standardUserClient, err = rb.client.AsUser(newUser)
+			require.NoError(rb.T(), err)
+			subSession := rb.session.NewSession()
+			defer subSession.Cleanup()
```

Comment:
I don't understand this part, why are we instantiating a new sub-session at the end of the `t.Run`?